src/main/host/syscall/type_formatting.rs

```diff
@@ -24,7 +48,7 @@ macro_rules! simple_display_impl {
                 match <$type>::try_from(self.reg).ok() {
                     Some(x) => write!(f, "{x}"),
                     // if the conversion to type T was unsuccessful, just show an integer
-                    None => write!(f, "{:#x} <invalid>", u64::from(self.reg)),
+                    None => fmt_int_with_suffix(f, u64::from(self.reg), "<invalid-type>"),
```

Comment:
What about "invalid-value" instead? The type should be correct, but the supplied value isn't valid for the type that the syscall takes.